flink-core/src/main/java/org/apache/flink/configuration/MemorySize.java

```diff
@@ -115,6 +115,27 @@ public String toString() {
 		return bytes + " bytes";
 	}
 
+	// ------------------------------------------------------------------------
+	//  Calculations
+	// ------------------------------------------------------------------------
+
+	public MemorySize add(MemorySize that) {
+		return new MemorySize(Math.addExact(this.bytes, that.bytes));
+	}
+
+	public MemorySize subtract(MemorySize that) {
+		return new MemorySize(Math.subtractExact(this.bytes, that.bytes));
+	}
+
+	public MemorySize multiply(double multiplier) {
+		checkArgument(multiplier >= 0, "multiplier must be >= 0");
+		double product = this.bytes * multiplier;
```

Comment:
MemorySize is essentially a size in bytes stored in long type. Converting from double type to long type should not be a problem. We might loose at most 1 byte in each multiply operation due to the rounding down error, which should be acceptable..

Reply:
Agreed. 
Besides, we could change the type of `product` to long, to avoid potential precision issue in next line. There is no use case to treat that value as double.

Reply:
Changing `product` to `long` could cause overflow.

Reply:
True, the error can be up to 9 bits and thus should not be ignored. I think it's a good idea to use BigDecimal for safety.